contrib/pkg/output/snapshot.go

```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
 	ListsToSync []ResourceList
 }
 
+// Options for writing resources of a given type
+type OutputOpts struct {
+	// error handler
+	ErrHandler ErrorHandler
+
+	// If provided, ensure the resource has been verified before adding it to snapshots
+	Verifier verifier.OutputResourceVerifier
+}
+
 // sync the output snapshot to local cluster storage.
 // only writes resources intended for the local cluster (with ClusterName == "")
 // Note that Syncing snapshots in this way adds the label
-func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, errHandler ErrorHandler) {
+func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, opts OutputOpts) {
 
+	var presentResources []ezkube.Object
 	for _, list := range s.ListsToSync {
 		listForLocalCluster := list.SplitByClusterName()[multicluster.LocalCluster]
 
+		for _, resource := range listForLocalCluster {
+			if opts.Verifier != nil {
+				gvk := resource.GetObjectKind().GroupVersionKind()
+				resourceRegistered, _ := opts.Verifier.VerifyServerResource(
+					multicluster.LocalCluster,
+					gvk,
+				)
+
+				if resourceRegistered {
+					presentResources = append(presentResources, resource)
+				}
+			}
+		}
+
 		resourcesForLocalCluster := ResourceList{
-			Resources:    listForLocalCluster,
+			Resources:    presentResources,
 			ListFunc:     list.ListFunc,
 			ResourceKind: list.ResourceKind,
 			StatusUpdate: list.StatusUpdate,
 		}
 
-		s.syncList(ctx, multicluster.LocalCluster, cli, resourcesForLocalCluster, errHandler)
+		s.syncList(ctx, multicluster.LocalCluster, cli, resourcesForLocalCluster, opts.ErrHandler)
 	}
 }
 
 // sync the output snapshot to storage across multiple clusters.
 // uses the object's ClusterName to determine the correct destination cluster.
-func (s Snapshot) SyncMultiCluster(ctx context.Context, mcClient multicluster.Client, errHandler ErrorHandler) {
-
+func (s Snapshot) SyncMultiCluster(ctx context.Context, mcClient multicluster.Client, opts OutputOpts) {
 	for _, list := range s.ListsToSync {
 		listsByCluster := list.SplitByClusterName()
 		// TODO(ilackarms): possible error case that we're ignoring here;
 		// we only write resources to clusters that are available to the multicluster client
 		// if the cluster is not available, we will not error (simply skip writing the resources here)
 		for _, cluster := range s.Clusters {
 			listForCluster := listsByCluster[cluster]
+			var presentResources []ezkube.Object
 
 			cli, err := mcClient.Cluster(cluster)
 			if err != nil {
 				for _, object := range listForCluster {
 					// send a write error for every object in the cluster
-					errHandler.HandleWriteError(object, err)
+					opts.ErrHandler.HandleWriteError(object, err)
 				}
 				continue
 			}
 
+			for _, resource := range listForCluster {
+				if opts.Verifier != nil {
+					gvk := resource.GetObjectKind().GroupVersionKind()
+					resourceRegistered, _ := opts.Verifier.VerifyServerResource(
```

Comment:
Can we either debug log the error here, or add a comment on why it's ok to ignore it?